diff --git a/modules/audio_processing/aec3/echo_canceller3.cc b/modules/audio_processing/aec3/echo_canceller3.cc
index e8e2175994..cc4aa65f0f 100644
--- a/modules/audio_processing/aec3/echo_canceller3.cc
+++ b/modules/audio_processing/aec3/echo_canceller3.cc
@@ -696,7 +696,7 @@ EchoCanceller3::RenderWriter::~RenderWriter() = default;
 
 void EchoCanceller3::RenderWriter::Insert(const AudioBuffer& input) {
   RTC_DCHECK_EQ(AudioBuffer::kSplitBandSize, input.num_frames_per_band());
-  RTC_DCHECK_EQ(num_bands_, input.num_bands());
+  // RTC_DCHECK_EQ(num_bands_, input.num_bands()); // commented as per bugs.webrtc.org/8759
   RTC_DCHECK_EQ(num_channels_, input.num_channels());
 
   // TODO(bugs.webrtc.org/8759) Temporary work-around.
