diff --git a/media/engine/webrtc_voice_engine.cc b/media/engine/webrtc_voice_engine.cc
index d1a22b5b03..366297bf41 100644
--- a/media/engine/webrtc_voice_engine.cc
+++ b/media/engine/webrtc_voice_engine.cc
@@ -1737,6 +1737,20 @@ bool WebRtcVoiceSendChannel::MuteStream(uint32_t ssrc, bool muted) {
         std::all_of(send_streams_.begin(), send_streams_.end(),
                     [](const auto& kv) { return kv.second->IsMuted(); });
 
+    // Because multiple WebRTCVoiceMediaChannels can be
+    // associated with a single AudioProcessing module,
+    // muting when all streams of this media channel
+    // are muted is incorrect, because streams on other
+    // channels may not be muted.  So, for now,
+    // we'll simply not disable audio processing when
+    // streams are muted.
+    //
+    // TODO(roxanneskelly)
+    // Track which WebRTCVoiceMediaChannels are all_muted
+    // and only call set_output_will_be_muted when all of them
+    // are muted.
+    is_all_muted = false;
+
     // Only mute the microphone if we're not in cleanup state
     // (i.e. if we have active send streams)
     webrtc::AudioProcessing* ap = engine()->apm();
