From 82a4fa0b2e4b6cc37a7c4dbb37d705f78405c187 Mon Sep 17 00:00:00 2001
From: Roxie Linden <roxie@lindenlab.com>
Date: Mon, 8 Jan 2024 20:09:24 -0800
Subject: [PATCH] upsample to 48khz for echo cancellation (for now)

---
 audio/audio_transport_impl.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/audio/audio_transport_impl.cc b/audio/audio_transport_impl.cc
index 42a81d5b4a..24de07d613 100644
--- a/audio/audio_transport_impl.cc
+++ b/audio/audio_transport_impl.cc
@@ -36,7 +36,9 @@ void InitializeCaptureFrame(int input_sample_rate,
                             size_t send_num_channels,
                             AudioFrame* audio_frame) {
   RTC_DCHECK(audio_frame);
-  int min_processing_rate_hz = std::min(input_sample_rate, send_sample_rate_hz);
+  //int min_processing_rate_hz = send_sample_rate_hzstd::min(input_sample_rate, send_sample_rate_hz);
+  // Second Life uses 48khz for spatial audio, so upsample to that for now.
+  int min_processing_rate_hz = send_sample_rate_hz;
   for (int native_rate_hz : AudioProcessing::kNativeSampleRatesHz) {
     audio_frame->sample_rate_hz_ = native_rate_hz;
     if (audio_frame->sample_rate_hz_ >= min_processing_rate_hz) {
-- 
2.43.0.windows.1

