diff --git a/modules/audio_device/win/core_audio_utility_win.cc b/modules/audio_device/win/core_audio_utility_win.cc
index dbfefa0428..a72443f40d 100644
--- a/modules/audio_device/win/core_audio_utility_win.cc
+++ b/modules/audio_device/win/core_audio_utility_win.cc
@@ -904,9 +904,9 @@ HRESULT SetClientProperties(IAudioClient2* client) {
   }
   AudioClientProperties props = {0};
   props.cbSize = sizeof(AudioClientProperties);
-  // Real-time VoIP communication.
+  // (roxanneskelly): Don't duck other audio sources.  Use Media instead of Communication
   // TODO(henrika): other categories?
-  props.eCategory = AudioCategory_Communications;
+  props.eCategory = AudioCategory_Media;
   // Hardware-offloaded audio processing allows the main audio processing tasks
   // to be performed outside the computer's main CPU. Check support and log the
   // result but hard-code `bIsOffload` to FALSE for now.
